File: JINSEOP/프로그래머스_WEEK8/2016년.cpp
#include <string>
#include <vector>
using namespace std;

string solution(int a, int b) {
    string answer = "";
    vector<string> days = { "SUN", "MON", "TUE", "WED", "THU", "FRI", "SAT" };
    switch (a)
    {
    case 1:
        answer = days[(b % 7 + 4)%7];
        break;
    case 2:
        answer = days[(b % 7) % 7];
        break;
    case 3:
        answer = days[(b % 7 + 1) % 7];
        break;
    case 4:
        answer = days[(b % 7 + 4) % 7];
        break;
    case 5:
        answer = days[(b % 7 + 6) % 7];
        break;
    case 6:
        answer = days[(b % 7 + 2) % 7];
        break;
    case 7:
        answer = days[(b % 7 + 4) % 7];
        break;
    case 8:
        answer = days[(b % 7) % 7];
        break;
    case 9:
        answer = days[(b % 7 + 3) % 7];
        break;
    case 10:
        answer = days[(b % 7 + 5) % 7];
        break;
    case 11:
        answer = days[(b % 7 + 1) % 7];
        break;
    case 12:
        answer = days[(b % 7 + 3) % 7];
        break;
    }

    return answer;
}

File: DONGUK/프로그래머스_WEEK 8/2016년.py
def solution(a, b):
    answer = ''
    
    DAYS = ["SUN","MON","TUE","WED","THU","FRI","SAT"]
    
    cal = [[] for _ in range(13)]
    
    ind = 5
    for i in range(1, 13):
        if i == 1 or i == 3 or i == 5 or i == 7 or i == 8 or i == 10 or i == 12:
            for _ in range(31):
                cal[i].append(DAYS[ind])
                ind += 1
                if ind > 6:
                    ind= ind % 7
                
        elif i == 2:
            for _ in range(29):
                cal[i].append(DAYS[ind])
                ind += 1
                if ind > 6:
                    ind = ind % 7
                
        else:
            for _ in range(30):
                cal[i].append(DAYS[ind])
                ind += 1
                if ind > 6:
                    ind = ind % 7
        
    answer = cal[a][b-1]
    return answer